<!--
  Arguments:
  lat   : Home latitude
  lon   : Home longitude
  alt   : Home altitude
  freq  : Update frequency of the PX4_msgs SensorGps message
  address : name or IP address of the Qualisys Mocap Server
  name : body-name, coming from Mocap system which need to be used
  vel: type of velocity information send by pseudo-GPS
    vel=0 : no velocity
    vel=1 : unfiltered velocity (estimate between current and previous mocap frames)
    vel=2 : filtered velocity (same as vel=1, but temporally filtered)

  example usage:
    ros2 launch mocap_pose mocap_pose.launch lat:=60.000 lon:=31.000 alt:=50.0
-->

<launch>
  <arg name="lat" default="61.50341"/>
  <arg name="lon" default="23.77509"/>
  <arg name="alt" default="110.0"/>
  <arg name="freq" default="10.0"/>
  <arg name="address" default="172.18.32.111"/>
  <arg name="name" default="$(env DEVICE_ALIAS)"/>
  <arg name="vel" default="1"/>


  <node name="mocap_pose" pkg="mocap_pose" exec="mocap_pose_node" namespace='$(env DRONE_DEVICE_ID)'>
    <param name="home_lat" value="$(var lat)"/>
    <param name="home_lon" value="$(var lon)"/>
    <param name="home_alt" value="$(var alt)"/>
    <param name="frequency" value="$(var freq)"/>
    <param name="server_address" value="$(var address)"/>
    <param name="body_name" value="$(var name)"/>
    <param name="velocity_type" value="$(var vel)"/>
  </node>

</launch>

